File: server/models/Commentmodel.js
const mongoose=require("mongoose");

const commentScheme=mongoose.Schema({
    commenttitle:{
        type:String,
        required:true,
        trim:true
    },
    postid:{
        type:String,
        required:true,
        trim:true
    },
    username:{
        type:String,
        required:true,
        trim: true,

        
    },
    userimage:{
        type:String,
        required:true,
        trim: true,

        
    },
    
});
const Comment=mongoose.model("Comment",commentScheme);
module.exports=Comment;

File: lib/models/comment_model.dart
import 'dart:convert';

// ignore_for_file: public_member_api_docs, sort_constructors_first
class CommentModel {
  final String commenttitle;
  final String username;
  final String userimage;
  final String postid;
  CommentModel({
    required this.commenttitle,
    required this.username,
    required this.userimage,
    required this.postid,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'commenttitle': commenttitle,
      'username': username,
      'userimage': userimage,
      'postid': postid,
    };
  }

  factory CommentModel.fromMap(Map<String, dynamic> map) {
    return CommentModel(
      commenttitle: map['commenttitle'] as String,
      username: map['username'] as String,
      userimage: map['userimage'] as String,
      postid: map['postid'] as String,
    );
  }

  String toJson() => json.encode(toMap());

  factory CommentModel.fromJson(String source) =>
      CommentModel.fromMap(json.decode(source) as Map<String, dynamic>);
}
